src/oauth2.rs

```diff
@@ -51,10 +48,10 @@ pub fn dc_get_oauth2_url(
     redirect_uri: impl AsRef<str>,
 ) -> Option<String> {
     if let Some(oauth2) = Oauth2::from_address(addr) {
-        set_config(
+        context.sql.set_config(
```

Comment:
here and also below i don't understand why set_config() and get_config() have been made more boilerplatish to call -- it's always "context.sql.get_config(context, ..." which seems like `get_config(context, ...` is more succinct.

Reply:
well the character amount is roughly the same:

- `get_config(context, &context.sql, ...)`

Reply:
in all cases you need to pass both `context.sql` and `context` somehow

Reply:
+1 for `context.sql.get_config()`